### PR TITLE
call createValidation before rollback. Fixes #72256

### LINKED ISSUE #72256: deployments/rollback not passed to validation
**What happened**:
I'm unable to configure the api-server through a `ValidatingWebhookConfiguration` to pass admission requests to my `ValidatingAdmissionWebhook` for the `deployments/rollback` resource. 

I'm adding the following rule to the `ValidatingWebhookConfiguration`
```
  - apiGroups:
    - extensions
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    resources:
    - deployments/rollback
```

When performing rollbacks, I'm not seeing any calls to the admission webhook.

**What you expected to happen**:

To receive validation requests when performing rollbacks.

**How to reproduce it (as minimally and precisely as possible)**:
Create a ValidatingAdmissionWebhook and perform a rollback request.

**Anything else we need to know?**:
Mentioned this on the #sig-api-machinery channel on Slack, and @liggitt suggested the error was that the `createValidation`-object never is used: https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/apps/deployment/storage/storage.go#L163-L183

**Environment**:
- Kubernetes version (use `kubectl version`):
1.12.3
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
Container Linux by CoreOS 1911.5.0 (Rhyolite)   4.14.84-coreos   docker://18.6.1
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by jhrv**
**What type of PR is this?**
/kind bug
/sig apps
/sig api-machinery

**What this PR does / why we need it**:
Ensures validation for rollback requests

**Which issue(s) this PR fixes**:

Fixes #72256 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE 

```release-note
`deployments/rollback` is now passed through validation/admission controllers
```

### CODE REVIEW

File: pkg/registry/apps/deployment/storage/storage.go [hunk 1]
```diff
@@ -166,6 +166,10 @@ func (r *RollbackREST) Create(ctx context.Context, obj runtime.Object, createVal
 		return nil, errors.NewBadRequest(fmt.Sprintf("not a DeploymentRollback: %#v", obj))
 	}
 
+	if err := createValidation(obj.DeepCopyObject()); err != nil {
```
Comment (yue9944882) [inline pkg/registry/apps/deployment/storage/storage.go]: obj ~.DeepCopyObject()~

Comment (jhrv) [reply pkg/registry/apps/deployment/storage/storage.go]: fixed

**Comment by yue9944882** [comment]
validating rollback here makes sense (i suppose?). but wiring into REST looks odd, why is rollback a creator instead of a connector at the beginning?  @kubernetes/sig-api-machinery-misc

File: pkg/registry/apps/deployment/storage/storage_test.go [hunk 2]
```diff
@@ -374,6 +379,31 @@ func TestEtcdCreateDeploymentRollback(t *testing.T) {
...
+	ctx := genericapirequest.WithNamespace(genericapirequest.NewContext(), namespace)
+
+	if _, err := storage.Deployment.Create(ctx, validNewDeployment(), rest.ValidateAllObjectFunc, &metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	_, err := rollbackStorage.Create(ctx, &rollback, denyCreateValidation, &metav1.CreateOptions{})
+
+	if err == nil {
+		t.Error("Expected admission error but got nothing")
```
Comment (yue9944882) [inline pkg/registry/apps/deployment/storage/storage_test.go]: >  	return fmt.Errorf("admission denied")

verify returned error

Comment (jhrv) [reply pkg/registry/apps/deployment/storage/storage_test.go]: fixed in https://github.com/kubernetes/kubernetes/pull/72271/commits/0293c9cc218931dd3568e93c394febad03179497

**Review by yue9944882 — APPROVED** [review]
LGTM please squash

/hold cancel

File: pkg/registry/apps/deployment/storage/storage.go (hunk 1 above)
Comment (liggitt) [inline pkg/registry/apps/deployment/storage/storage.go]: the generic registry checks creationValidation for nil... would probably be good to do the same here

File: pkg/registry/core/serviceaccount/storage/token.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ var gvk = schema.GroupVersionKind{
 }
 
 func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	if err := createValidation(obj); err != nil {
+	if err := createValidation(obj.DeepCopyObject()); err != nil {
```
Comment (liggitt) [inline pkg/registry/core/serviceaccount/storage/token.go]: add nil check here as well

File: pkg/registry/apps/deployment/storage/storage.go (hunk 1 above)
Comment (jhrv) [reply pkg/registry/apps/deployment/storage/storage.go]: added

File: pkg/registry/core/serviceaccount/storage/token.go (hunk 3 above)
Comment (jhrv) [reply pkg/registry/core/serviceaccount/storage/token.go]: added

**Pull request merged**